src/librustc_mir/transform/instrument_coverage.rs

```diff
@@ -77,17 +79,24 @@ fn const_int_operand<'tcx>(
     tcx: TyCtxt<'tcx>,
     span: Span,
     ty: Ty<'tcx>,
-    val: u128,
+    val: Scalar,
 ) -> Operand<'tcx> {
-    let param_env_and_ty = ty::ParamEnv::empty().and(ty);
-    let size = tcx
-        .layout_of(param_env_and_ty)
-        .unwrap_or_else(|e| panic!("could not compute layout for {:?}: {:?}", ty, e))
-        .size;
+    debug_assert!({
```

Comment:
This might be redundant with MIR typeck, not sure. Can't hurt, though.

Reply:
Yeah, I assume there are more checks, but this debug-only check makes it easier to pinpoint where the problem is introduced.